LegionBoard: fix bug with course being "null"

AltNico opened the pull request:
When upgrading LegionBoard Heart from `0.1.x` to `0.2+`, all existing changes get the course "null". Mistakenly, I thought the JSONObject would contain a "real" null, but the null is given as a string.

The following IOException is fixed with this commit:

```
java.io.IOException: Change references a course but courses are empty.
    at me.vertretungsplan.parser.LegionBoardParser.getSubstitution(LegionBoardParser.java:201)
    at me.vertretungsplan.parser.LegionBoardParser.parseLegionBoard(LegionBoardParser.java:160)
    at me.vertretungsplan.parser.LegionBoardParser.getSubstitutionSchedule(LegionBoardParser.java:69)
```

johan12345 commented:
Have you tried something like

```
change.isNull("course")
```

instead? Or is the JSON that Heart returns really `"course": "null"` instead of `"course": null`?

AltNico commented:
Both is working. Should I push a new commit?

> Or is the JSON that Heart returns really `"course": "null"` instead of `"course": null`?

It is returning `"course": null`.

johan12345 commented:
Yes, I think using `isNull` is better as it does not rely on the (seemingly undocumented) behavior of `JSONObject`'s `getString()` returning `"null"` when the value is `null`.

AltNico committed:
LegionBoard: fix bug with course being "null"

When upgrading LegionBoard Heart from 0.1.x to 0.2+, all existing changes get the course "null". Mistakenly, I thought the JSONObject would contain a "real" null, but the null is given as a string.

The following IOException is fixed with this commit:
java.io.IOException: Change references a course but courses are empty.
	at me.vertretungsplan.parser.LegionBoardParser.getSubstitution(LegionBoardParser.java:201)
	at me.vertretungsplan.parser.LegionBoardParser.parseLegionBoard(LegionBoardParser.java:160)
	at me.vertretungsplan.parser.LegionBoardParser.getSubstitutionSchedule(LegionBoardParser.java:69)

AltNico commented:
Done.

johan12345 commented:
Thanks! I created a new release `1.0.0-beta19`. It is now also running on the vertretungsplan.me server, so the app should start to work again.

AltNico commented:
Thank you for your quick feedback :D
